core/src/main/java/io/questdb/PropServerConfiguration.java

```diff
@@ -497,6 +527,21 @@ private String getString(Properties properties, String key, String defaultValue)
         return value;
     }
 
+    private TimestampFormat getTimestampFormat(Properties properties, String key, String defaultPattern) throws ServerConfigurationException {
+        String pattern = properties.getProperty(key);
+        if (null == pattern) {
+            pattern = defaultPattern;
+        }
+        DateFormatCompiler compiler = new DateFormatCompiler();
+        if (null != pattern) {
+            if (pattern.contains(new StringBuilder().append(Files.SEPARATOR).toString()) || pattern.startsWith(".")) {
```

Comment:
I feel these checks are unnecessary. The config comes from file on disk, so attacker would have already gotten access to file system.

Additionally there are methods like `Chars.contains()` and others, which would let you check if CharSequence contains `\` without creating a string.